apps/studio/electron/preload/webview/dom.ts

```diff
@@ -1,10 +1,62 @@
 import { ipcRenderer } from 'electron';
+import { EditorAttributes } from '@onlook/models/constants';
 import { getOrAssignUuid } from './elements/helpers';
 import { WebviewChannels } from '@onlook/models/constants';
 import { getUniqueSelector, isValidHtmlElement } from '/common/helpers';
 import type { LayerNode } from '@onlook/models/element';
 
+function markDynamicElements(root: HTMLElement) {
+    const containers = root.querySelectorAll(`[${EditorAttributes.DATA_ONLOOK_ID}]`);
+
+    containers.forEach((container) => {
+        const children = Array.from(container.children);
+        if (children.length < 2) {
+            return;
+        }
+
+        const groups = children.reduce(
+            (acc, child) => {
+                if (!isValidHtmlElement(child)) {
+                    return acc;
+                }
+
+                const signature = [
+                    child.tagName,
+                    child.className,
+                    child.getAttribute(EditorAttributes.DATA_ONLOOK_ID),
+                    Array.from(child.attributes)
+                        .filter((attr) => attr.name.startsWith('data-') || attr.name === 'key')
+                        .map((attr) => attr.name)
+                        .sort()
+                        .join(','),
+                ].join('|');
+
+                if (!acc[signature]) {
+                    acc[signature] = [];
+                }
+                acc[signature].push(child);
+                return acc;
+            },
+            {} as Record<string, Element[]>,
+        );
+
+        // Mark elements that is part of a map/iteration, dynamic
+        Object.values(groups).forEach((group) => {
+            if (group.length > 1) {
+                group.forEach((element) => {
+                    if (element instanceof HTMLElement) {
+                        element.setAttribute(EditorAttributes.DATA_ONLOOK_DYNAMIC, 'true');
```

Comment:
I would think that if a `DATA_ONLOOK_DYNAMIC_TYPE ` exist at all, it would be dynamic. So this may be redundant.

Reply:
True. I initially started with DATA_ONLOOK_DYNAMIC only, but then decided to go further by trying to detect the type. You are correct - I will remove DATA_ONLOOK_DYNAMIC since it's now redundant.